### PR TITLE
Features/reject verify persist

### PR DESCRIPTION
**Pull request by SamirHassoun**
1) Converted PersistWorkflowInstanceDataEvent to use WorkflowStage enum for FromActivity and ToActivity
2) Use Newtonsoft.Json to handle serialise\deserialise of NSB messages
3) Use WorkflowStage enum instead of hard-coded stages
4) Tidied the code of PersistWorkflowInstanceDataEvent when handling different stages
5) Persist Verify data to Assess when rejecting

### CODE REVIEW

File: src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs [hunk 1]
```diff
@@ -103,32 +91,59 @@ public async Task Handle(PersistWorkflowInstanceDataEvent message, IMessageHandl
...
-            int workflowInstanceId, string fromActivityName, WorkflowInstance workflowInstance)
+        private void ValidateK2TaskForSignOff(PersistWorkflowInstanceDataEvent message, K2TaskData k2Task)
         {
-            LogContext.PushProperty("PersistWorkflowDataToAssess", nameof(PersistWorkflowDataToAssess));
-            LogContext.PushProperty("ProcessId", processId);
-            LogContext.PushProperty("WorkflowInstanceId", workflowInstanceId);
-            LogContext.PushProperty("FromActivityName", fromActivityName);
+            if (k2Task != null)
+            {
+                // if task as at Completed, then we expect k2Task to be null
```
Comment (Rossall82) [inline src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs]:
```suggestion
                // if task is at Completed, then we expect k2Task to be null
```

File: src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs [hunk 2]
```diff
@@ -103,32 +91,59 @@ public async Task Handle(PersistWorkflowInstanceDataEvent message, IMessageHandl
...
+            }
+        }
 
-            _logger.LogInformation("Entering {PersistWorkflowDataToAssess} with processId: {ProcessId}; " +
-                                   "workflowInstanceId: {WorkflowInstanceId}; and fromActivityName: {FromActivityName}.");
+        private void ValidateK2TaskForAssessAndVerify(PersistWorkflowInstanceDataEvent message, K2TaskData k2Task)
+        {
+            if (k2Task == null)
+            {
+                _logger.LogError("Failed to get data for K2 Task at stage with ProcessId {ProcessId}");
```
Comment (Rossall82) [inline src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs]: Can we add what stage we're expecting the task to be at here (both the log entry and Exception msg)?

File: src/WorkflowCoordinator/WorkflowCoordinator/Program.cs [hunk 3]
```diff
@@ -84,12 +83,12 @@ private static async Task Main(string[] args)
                     services.AddDbContext<WorkflowDbContext>((serviceProvider, options) =>
                         options.UseSqlServer(connection));
 
-                    if (isLocalDebugging)
-                    {
-                        using var sp = services.BuildServiceProvider();
-                        using var context = sp.GetRequiredService<WorkflowDbContext>();
-                        TestWorkflowDatabaseSeeder.UsingDbContext(context).PopulateTables().SaveChanges();
-                    }
+                    //if (isLocalDebugging)
```
Comment (Rossall82) [inline src/WorkflowCoordinator/WorkflowCoordinator/Program.cs]: What's the crack here?

File: src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs (hunk 1 above)
Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs]: Resolved, thanks

File: src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs (hunk 2 above)
Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/Handlers/PersistWorkflowInstanceDataEventHandler.cs]: Resolved, thanks

File: src/WorkflowCoordinator/WorkflowCoordinator/Program.cs (hunk 3 above)
Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/Program.cs]: Resolved, thanks

**Pull request merged**